src/components/modal/Modal.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { ReactNode, RefObject, useEffect, useState } from 'react';
+import ReactDOM from 'react-dom';
+import { IconAlertCircle, IconCross } from 'hds-react';
+import { PrimaryButton, SecondaryButton } from '../button/Button';
+import './Modal.scss';
+
+const ModalBackDrop = (): JSX.Element => <div className="modal-backdrop" />;
+const ModalWrapper = ({
+  id,
+  ariaLabelledBy,
+  children,
+}: {
+  id: string;
+  ariaLabelledBy: string;
+  children: ReactNode;
+}): JSX.Element => (
+  <div
+    role="alertdialog"
+    id={id}
+    className="modal-wrapper"
+    aria-labelledby={ariaLabelledBy}
+    aria-modal="true">
+    {children}
+  </div>
+);
+
+type UseModalProps = {
+  isOpen: boolean;
+  toggle: () => void;
+};
+
+export const useModal = (): UseModalProps => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const toggle = (): void => setIsOpen(!isOpen);
+  return {
+    isOpen,
+    toggle,
+  };
+};
+
+export function ConfirmationModal({
+  title,
+  text,
+  confirmText,
+  onConfirm,
+  isOpen,
+  close,
+}: {
+  title: string;
+  text: string | ReactNode;
+  confirmText: string;
+  onConfirm: () => void;
+  isOpen: boolean;
+  close: () => void;
+}): JSX.Element | null {
+  const titleId = 'confirmation-modal-title';
+  const buttonRef: RefObject<HTMLButtonElement> = React.createRef();
+
+  const onClose = (): void => {
+    close();
+  };
+
+  const onKeyDown = (event: KeyboardEvent): void => {
+    if (event.key === 'Escape') {
+      onClose();
+    }
+  };
+
+  useEffect(() => {
+    if (isOpen) {
+      document.body.classList.add('modal-visible');
```

Comment:
This feels like not a react way of doing dom manipulation but I guess it is the only good way to do it in this situation 🤔

Reply:
There might be some other way but this was the easiest 😁